std-bits/table/src/main/java/org/enso/table/data/column/storage/StringStorage.java

```diff
@@ -70,12 +73,24 @@ public Builder createDefaultBuilderOfSameType(int capacity) {
     return new StringBuilder(capacity);
   }
 
+  @Override
+  public Storage<?> cast(StorageType targetType, CastProblemBuilder castProblemBuilder) {
+    return switch (targetType) {
+      case AnyObjectType any -> new MixedStorageFacade(this);
+      case TextType textType -> adapt(this, textType);
+      default -> throw new IllegalStateException("Conversion of StringStorage to " + targetType + " is not supported");
```

Comment:
should we report use `parse`?

Reply:
We do. This `IllegalStateException` should never be reached - the calling Enso code already checks if the conversion is allowed or not.